src/shared/bodies/base_body.cpp

```diff
@@ -36,6 +36,12 @@ BoundingBox SPHBody::getSPHSystemBounds()
     return sph_system_.system_domain_bounds_;
 }
 //=================================================================================================//
+int SPHBody::getNewBodyPartID()
```

Comment:
Checking this ID seems to be better and easier than using the checkNotFar.  
But I guess there is a potential bug for this, if the .tag_buffer_particles.exec(); is in front of the deletion.
Since the outflow particles would not be tagged as the buffer particle.

Reply:
Yes. Tagging need to done at initial condition and immediately after cell linked list is done.